lib/delay-execution.yml

```diff
@@ -14,17 +16,23 @@ rule:
       - B5F85C26D7AA5A1FB4AF5821B6B5AB9B:0x402FA6
   features:
     - or:
-      - api: kernel32.Sleep
-      - api: kernel32.SleepEx
-      - api: kernel32.WaitForSingleObject
-      - api: kernel32.SignalObjectAndWait
-      - api: kernel32.WaitForSingleObjectEx
-      - api: kernel32.WaitForMultipleObjects
-      - api: kernel32.WaitForMultipleObjectsEx
-      - api: kernel32.RegisterWaitForSingleObject
-      - api: WaitOnAddress
-      - api: user32.MsgWaitForMultipleObjects
-      - api: user32.MsgWaitForMultipleObjectsEx
-      - api: NtDelayExecution
-      - api: KeWaitForSingleObject
-      - api: KeDelayExecutionThread
+      - function:
+        - and:
+          - basic block:
+            - or:
+              - api: kernel32.Sleep
+              - api: kernel32.SleepEx
+              - api: kernel32.WaitForSingleObject
+              - api: kernel32.SignalObjectAndWait
+              - api: kernel32.WaitForSingleObjectEx
+              - api: kernel32.WaitForMultipleObjects
+              - api: kernel32.WaitForMultipleObjectsEx
+              - api: kernel32.RegisterWaitForSingleObject
+              - api: WaitOnAddress
+              - api: user32.MsgWaitForMultipleObjects
+              - api: user32.MsgWaitForMultipleObjectsEx
+              - api: NtDelayExecution
+              - api: KeWaitForSingleObject
+              - api: KeDelayExecutionThread
+      - import: signal
```

Comment:
`signal` is used to configure the handling of many types of notifications, beyond sleeping (i think, at least as documented here: https://man7.org/linux/man-pages/man2/signal.2.html). so, i think this portion needs more logic to specify the specific type of signal that is being configured to effect a delay.